### PR TITLE
Fixed a bunch of test262 panics (#817)

### PR DESCRIPTION
**Pull request by Razican**
This PR works towards a panic-free boa. It fixes:
 - Panics in `RegExp`
 - Wrong `delete` argument
 - `parseInt()` panics if radix was out of bounds.
 - Spec-compliant `parseInt()` function.

I also took the opportunity to update the dependencies and the test262 submodule, and fix a couple of clippy warnings in tests, along with a documentation link.

### CODE REVIEW

File: boa/src/builtins/number/mod.rs [hunk 1]
```diff
@@ -651,54 +651,102 @@ impl Number {
...
-                                Ok(Value::integer(i))
-                            } else {
-                                // String can't be parsed.
-                                Ok(Value::from(f64::NAN))
-                            };
-                        } else {
-                            radix = 10
-                        };
-                    }
+            // 5. If S is not empty and the first code unit of S is the code unit 0x002B (PLUS SIGN) or the code unit 0x002D (HYPHEN-MINUS), remove the first code unit from S.
```
Comment (RageKnify) [inline boa/src/builtins/number/mod.rs:670]: Here

File: boa/src/builtins/number/mod.rs [hunk 2]
```diff
@@ -651,54 +651,102 @@ impl Number {
...
+            if strip_prefix
+                && var_s.len() >= 2
+                && (var_s.starts_with("0x") || var_s.starts_with("0X"))
+            {
+                var_s = var_s.split_at(2).1;
+
+                var_r = 16;
             }
+
+            // 11. If S contains a code unit that is not a radix-R digit, let end be the index within S of the first such code unit; otherwise, let end be the length of S.
```
Comment (RageKnify) [inline boa/src/builtins/number/mod.rs:713]: Here

File: boa/src/builtins/number/mod.rs [hunk 3]
```diff
@@ -651,54 +651,102 @@ impl Number {
...
+
+            // 12. Let Z be the substring of S from 0 to end.
+            let var_z = var_s.split_at(end).0;
+
+            // 13. If Z is empty, return NaN.
+            if var_z.is_empty() {
+                return Ok(Value::nan());
+            }
+
+            // 14. Let mathInt be the integer value that is represented by Z in radix-R notation, using the letters A-Z and a-z for digits with values 10 through 35. (However, if R is 10 and Z contains more than 20 significant digits, every significant digit after the 20th may be replaced by a 0 digit, at the option of the implementation; and if R is not 2, 4, 8, 10, 16, or 32, then mathInt may be an implementation-approximated value representing the integer value that is represented by Z in radix-R notation.)
```
Comment (RageKnify) [inline boa/src/builtins/number/mod.rs:728]: And here

**Review by RageKnify — CHANGES_REQUESTED** [review]
A few of the comments citing the spec are really long, I think it'd be best to split into multiple lines.
The rest looks great.

**Comment by Razican** [comment]
> A few of the comments citing the spec are really long, I think it'd be best to split into multiple lines.
> The rest looks great.

Should be fixed now :)

**Review by RageKnify — APPROVED** [review]

**Review by HalidOdat — APPROVED** [review]

**Pull request merged**